File: mycollab-services-pro/src/main/java/com/mycollab/pro/module/project/dao/RiskMapperExt.java
package com.mycollab.pro.module.project.dao;

import com.mycollab.db.persistence.IMassUpdateDAO;
import com.mycollab.db.persistence.ISearchableDAO;
import com.mycollab.module.project.domain.Risk;
import com.mycollab.module.project.domain.SimpleRisk;
import com.mycollab.module.project.domain.criteria.RiskSearchCriteria;

/**
 * @author MyCollab Ltd
 * @since 1.0.0
 */
public interface RiskMapperExt extends ISearchableDAO<RiskSearchCriteria>, IMassUpdateDAO<Risk, RiskSearchCriteria> {

    SimpleRisk findRiskById(int riskId);
}

File: mycollab-services-pro/src/main/java/com/mycollab/pro/module/project/dao/RiskMapperExt.kt
package com.mycollab.pro.module.project.dao

import com.mycollab.db.persistence.IMassUpdateDAO
import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.module.project.domain.Risk
import com.mycollab.module.project.domain.SimpleRisk
import com.mycollab.module.project.domain.criteria.RiskSearchCriteria

/**
 * @author MyCollab Ltd
 * @since 1.0.0
 */
interface RiskMapperExt : ISearchableDAO<RiskSearchCriteria>, IMassUpdateDAO<Risk, RiskSearchCriteria> {

    fun findRiskById(riskId: Int): SimpleRisk?
}
